javascript/react/introduction-to-hooks/state-hooks.md

```diff
@@ -0,0 +1,136 @@
+---
+author: kapnobatai136
+
+levels:
+
+  - medium
+  
+aspects:
+
+  - new
+
+  - workout
+
+  - deep
+
+type: normal
+
+category: how to
+
+links:
+  - '[state-hooks](https://reactjs.org/docs/hooks-state.html){documentation}'
+
+---
+
+# State hooks
+
+---
+## Content
+
+Let's start off by looking at how you would usually initialize a state in a class component:
+
+```js
+class Enki extends React.Component(
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: "Enki"
+    };
+  }
+)
+```
+
+But what if you wanted to use functional components? In a function component you can't use the `this` keyword, nor can you assign or read `this.state`. This is where hooks come in:
```

Comment:
Technically, you can use `this` in any JS function, it's an inherent binding.

Maybe just say

```suggestion
But what if you wanted to use functional components? In a function component you can't assign or read `this.state`. This is where hooks come in:
```